woocommerce/woocommerce
# Add additional unit tests for Store API checkout flow

### Changes proposed in this Pull Request:

Added the following unit tests:
- Ensure that orders can be placed with coupons.
- Ensure that orders cannot be placed with coupons over their usage limit.
- Ensure that orders cannot be placed with un-owned coupons. (e.g. email restriction)
- Ensure that orders cannot be placed with out of stock items.
- Ensure that orders cannot be placed with invalid payment methods.
- Ensure that orders can be placed with virtual products.


Closes #45584



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check CI passes.
2. Make sure the expected status codes in test match what you'd expect in the situation.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
